### PR TITLE
[5.1] Count is already known

### PR DESCRIPTION
**Pull request by likerRr**
Hello! I think we can eliminate the `count()` inside the `if` condition

### CODE REVIEW

File: src/Illuminate/Database/Migrations/Migrator.php [hunk 1]
```diff
@@ -188,7 +188,7 @@ public function reset($pretend = false)
         if (count($migrations) == 0) {
             $this->note('<info>Nothing to rollback.</info>');
 
-            return count($migrations);
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Migrations/Migrator.php]: Why not just asign a variable further up?

Comment (GrahamCampbell) [inline src/Illuminate/Database/Migrations/Migrator.php]: Then reference that instead out counting again further bellow.

**Comment by ArjanSchouten** [comment]
I think it should be this:
```php
    public function rollback($pretend = false)
    {
        $this->notes = [];
        // We want to pull in the last batch of migrations that ran on the previous
        // migration operation. We'll then reverse those migrations and run each
        // of them "down" to reverse the last migration "operation" which ran.
        $migrations = $this->repository->getLast();

        $migrationsCount = count($migrations);

        if ($migrationsCount == 0) {
            $this->note('<info>Nothing to rollback.</info>');

            return $migrationsCount;
        }
        // We need to reverse these migrations so that they are "downed" in reverse
        // to what they run on "up". It lets us backtrack through the migrations
        // and properly reverse the entire database schema operation that ran.
        foreach ($migrations as $migration) {
            $this->runDown((object) $migration, $pretend);
        }

        return $migrationsCount;
    }
```

**Comment by likerRr** [comment]
Yeah, great idea, also may be add `else` to avoid double `return $migrationsCount;`? What do you think?

```php
    public function rollback($pretend = false)
    {
        $this->notes = [];
        // We want to pull in the last batch of migrations that ran on the previous
        // migration operation. We'll then reverse those migrations and run each
        // of them "down" to reverse the last migration "operation" which ran.
        $migrations = $this->repository->getLast();

        $migrationsCount = count($migrations);

        if ($migrationsCount == 0) {
            $this->note('<info>Nothing to rollback.</info>');
        } else {
            // We need to reverse these migrations so that they are "downed" in reverse
            // to what they run on "up". It lets us backtrack through the migrations
            // and properly reverse the entire database schema operation that ran.
            foreach ($migrations as $migration) {
                $this->runDown((object) $migration, $pretend);
            }
        }

        return $migrationsCount;
    }

**Comment by ArjanSchouten** [comment]
Can't see why that's needed. If you leave the return where it was it's clear that it's returning 0. Doesn't really matter IMHO

**Comment by likerRr** [comment]
@ArjanSchouten well, agree

**Comment by ArjanSchouten** [comment]
It's more likely to be merged when you squash this to one commit. Why have you changed the equals sign to identical sign?

**Comment by GrahamCampbell** [comment]
TBH, this could be dried up too.

**Comment by likerRr** [comment]
@GrahamCampbell sorry, what do you mean?

**Comment by GrahamCampbell** [comment]
There is repeated code in here. This would be better:

```php
    public function rollback($pretend = false)
    {
        $this->notes = [];

        // We want to pull in the last batch of migrations that ran on the previous
        // migration operation. We'll then reverse those migrations and run each
        // of them "down" to reverse the last migration "operation" which ran.
        $migrations = $this->repository->getLast();

        $count = count($migrations);

        if ($count === 0) {
            $this->note('<info>Nothing to rollback.</info>');
        } else {
            // We need to reverse these migrations so that they are "downed" in reverse
            // to what they run on "up". It lets us backtrack through the migrations
            // and properly reverse the entire database schema operation that ran.
            foreach ($migrations as $migration) {
                $this->runDown((object) $migration, $pretend);
            }
        }

        return $count;
    }
```

**Comment by GrahamCampbell** [comment]
`$count` is a better name imo. It's obvious what it is - no need to make the name long.

**Comment by likerRr** [comment]
It's my first contribution experience, thank you for help

**Comment by taylorotwell** [comment]
It's not broken so I'm hesitant to merge a large fix to not actually fix a bug. :/

**Comment by franzliedke** [comment]
@taylorotwell To be honest, the diff makes the change look much larger than it is. Try the [side-by-side view](https://github.com/laravel/framework/pull/9674/files?diff=split). :)

**Comment by taylorotwell** [comment]
Yeah that helps.

**Pull request merged**